src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1492,8 +1492,13 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_WriteThreadNetw
             otOperationalDataset activeDataset;
             otError otErr = otDatasetGetActive(mOTInst, &activeDataset);
             VerifyOrExit(otErr == OT_ERROR_NONE, err = MapOpenThreadError(otErr));
-            uint64_t activeTimestamp = activeDataset.mPendingTimestamp;
-            err                      = encoder.Encode(activeTimestamp);
+#if OPENTHREAD_API_VERSION >= 219
+            uint64_t activeTimestamp = (activeDataset.mActiveTimestamp.mSeconds << 16) |
+                (activeDataset.mActiveTimestamp.mTicks << 1) | activeDataset.mActiveTimestamp.mAuthoritative;
```

Comment:
Why?  The spec is not super-clear on what this attribute means, probably because it assumes the old API surface, but why would we be including mAuthoritative in there?

Reply:
Thread 1.1.1 Section 8.10.1.13 states:

> A 1-bit flag that indicates the time was obtained from an authoritative source: either NTP (Network Time Protocol), GPS (Global Positioning System), cell network, or other method. Note that the U bit is the least significant bit when the entire timestamp is represented as a single 64-bit number.